### PR TITLE
[DOC]Add missing functions to API docs Index

### LINKED ISSUE #871: [DOC] Add factorize_columns and drop_constant_columns to the API index
- `factorize_columns` and `drop_constant_columns` are missing from the API index. This functions were created in #850 and #852 respectively.

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- Add `drop_constant_columns` and `factorize_columns` to the API

**This PR resolves #871 .**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [ ] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    functions: test for general functions
+    biology: tests for biology
+    chemistry: tests for chemistry
+    finance: tests for finance
+    utils: utility tests
+    engineering: tests for engineering
+    ml: tests for machine learning
+    spark_functions: tests for pyspark functions
+    xarray: tests for xarray functions
+    timeseries: tests for timeseries
```
Comment (ericmjl) [inline pytest.ini]: @samukweku I think this one is not necessary as the configurations for pytest are present in the `pyproject.toml` file.

**Review by ericmjl — CHANGES_REQUESTED** [review]
Only one small request, @samukweku - our config for pytest is in pyproject.toml!

Other than that, I think we're good to go :smile:.

File: pytest.ini (hunk 1 above)
Comment (samukweku) [reply pytest.ini]: So I can ignore @ericmjl

Comment (ericmjl) [reply pytest.ini]: That's right, I think we should delete `pytest.ini` - could you instead combine the contents of line 3 (`addopts`) with the addopts in `pyproject.toml`?

**Review by ericmjl — APPROVED** [review]

**Comment by ericmjl** [comment]
This being a docs PR, it's sufficient to have one review on it. I think we need to be a bit more conscientious with code reviews. Therefore, I merged with only myself as an official reviewer.

Thanks @samukweku! :tada:

**Pull request merged**